src/amqpvalue.c

```diff
@@ -6550,13 +6611,21 @@ static int internal_decoder_decode_bytes(INTERNAL_DECODER_DATA* internal_decoder
                                     uint32_t i;
 
                                     internal_decoder_data->decode_to_value->value.map_value.pair_count /= 2;
+                                    size_t malloc_size = safe_multiply_size_t((size_t)internal_decoder_data->decode_to_value->value.map_value.pair_count, 2);
+                                    malloc_size = safe_multiply_size_t(sizeof(AMQP_MAP_KEY_VALUE_PAIR), malloc_size);
+
                                     if (internal_decoder_data->decode_to_value->value.map_value.pair_count > MAX_AMQPVALUE_ITEM_COUNT)
                                     {
                                         LogError("AMQP list map count exceeded MAX_AMQPVALUE_ITEM_COUNT");
                                         result = MU_FAILURE;
                                     }
+                                    else if (malloc_size == SIZE_MAX)
```

Comment:
Why do we check against `SIZE_MAX` here but not other places?  Shouldn't the `safe_multiple_size_t` protect us from overflow and then on off-shot we ended up with `SIZE_MAX` we'd rely on the malloc() call to fail but otherwise be OK knowing we ourselves hadn't over-flown?

Reply:
It's a known error state, so probably should not call malloc, also it gives more useful trace statement.